src/mutate/drawDefinitions/positionGovernor/qualifierProgression.ts

```diff
@@ -0,0 +1,163 @@
+import { findExtension } from '@Acquire/findExtension';
+import { findStructure } from '@Acquire/findStructure';
+import { qualifierDrawPositionAssignment } from '@Assemblies/governors/drawsGovernor';
+import { POSITION, QUALIFYING, WINNER } from '@Constants/drawDefinitionConstants';
+import {
+  MISSING_QUALIFIED_PARTICIPANTS,
+  NO_DRAW_POSITIONS_AVAILABLE_FOR_QUALIFIERS,
+} from '@Constants/errorConditionConstants';
+import { TALLY } from '@Constants/extensionConstants';
+import { BYE } from '@Constants/matchUpStatusConstants';
+import { POLICY_TYPE_POSITION_ACTIONS } from '@Constants/policyConstants';
+import { SUCCESS } from '@Constants/resultConstants';
+import { decorateResult } from '@Functions/global/decorateResult';
+import { getPositionAssignments, structureAssignedDrawPositions } from '@Query/drawDefinition/positionsGetter';
+import { isCompletedStructure } from '@Query/drawDefinition/structureActions';
+import { getAppliedPolicies } from '@Query/extensions/getAppliedPolicies';
+import { getAllStructureMatchUps } from '@Query/matchUps/getAllStructureMatchUps';
+import { getSourceStructureIdsAndRelevantLinks } from '@Query/structure/getSourceStructureIdsAndRelevantLinks';
+import { randomPop } from '@Tools/arrays';
+import { definedAttributes } from '@Tools/definedAttributes';
+import { ResultType } from '@Types/factoryTypes';
+import { DrawDefinition, Event, Structure, Tournament } from '@Types/tournamentTypes';
+
+interface QualifierProgressionArgs {
+  drawDefinition: DrawDefinition;
+  event: Event;
+  mainStructureId: string;
+  tournamentRecord: Tournament;
+}
+
+export function qualifierProgression({
+  drawDefinition,
+  event,
+  mainStructureId,
+  tournamentRecord,
+}: QualifierProgressionArgs): ResultType {
```

Comment:
you could usefully add:

  const paramsCheck = checkRequiredParameters(params, [{ [DRAW_DEFINITION]: true, [EVENT]: true, [TOURNAMENT_RECORD]: true }]);
  if (paramsCheck.error) return paramsCheck;